### PR TITLE
Optimize security settings creation with custom filter

### PR DESCRIPTION
**Pull request by cihlamar**
AMW issue: https://redhat.atlassian.net/browse/AMW-506
GitHub issue: https://github.com/ansible-middleware/amq/issues/248

Replace expensive loop pattern using subelements() and combine() with efficient filter plugin. 
Provides significant performance improvement for deployments with large numbers of roles and permissions.

- Add create_security_matches filter plugin
- Update user_roles.yml to use new filter
- Update validate_config.yml to use new filter

### CODE REVIEW

File: plugins/filter/create_security_matches.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        )
+
+    result = {}
+
+    for role in amq_broker_roles:
+        role_name = role.get('name')
+        if not role_name:
+            continue
+
+        match_pattern = role.get('match') or '#'
```
Comment (RanabirChakraborty) [inline plugins/filter/create_security_matches.py:70]: The original code references item.0['match'] but the filter uses role.get('match'). If 'match' key doesn't exist, the original used default('#') while yours uses role.get('match') or '#' - these behave identically for missing keys, but differently for explicit None or empty string values.

File: plugins/filter/create_security_matches.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    if not isinstance(amq_broker_roles, list):
+        raise AnsibleFilterError(
+            'create_security_matches requires a list of roles'
+        )
+
+    result = {}
+
+    for role in amq_broker_roles:
+        role_name = role.get('name')
+        if not role_name:
```
Comment (RanabirChakraborty) [inline plugins/filter/create_security_matches.py:65]: Consider logging a warning or raising an error for invalid input to aid debugging

File: plugins/filter/create_security_matches.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
```
Comment (RanabirChakraborty) [inline plugins/filter/create_security_matches.py:2]: I guess it's Python 2.7 compatible, check it once. We are having python 3.12 support need to script accordingly.

**Review by RanabirChakraborty — CHANGES_REQUESTED** [review]
Apart from the below review if possible try to add a test for this edge case.

**Review by cihlamar — COMMENTED** [review]
review processed

**Review by RanabirChakraborty — APPROVED** [review]
LGTM, @cihlamar just rebase both the commit into one.

**Comment by cihlamar** [comment]
> LGTM, @cihlamar just rebase both the commit into one.

Rebased into single commit.
